src/main/java/io/antmedia/AntMediaApplicationAdapter.java

```diff
@@ -1494,7 +1524,49 @@ private void updateAppSettingsBean(AppSettings appSettings, AppSettings newSetti
 		appSettings.setIpFilterEnabled(newSettings.isIpFilterEnabled());
 		appSettings.setJwtControlEnabled(newSettings.isJwtControlEnabled());
 		appSettings.setJwtSecretKey(newSettings.getJwtSecretKey());
-		
+
+
+		appSettings.setS3RecordingEnabled(newSettings.isS3RecordingEnabled());
+
+		if (appSettings.isS3RecordingEnabled()) {
+			appSettings.setS3AccessKey(newSettings.getS3AccessKey());
+			if (appSettings.getS3AccessKey().isEmpty()) {
+			   logger.warn("S3 recording is enabled but access key is not given for app:{} ", getScope().getName());
+			}
+			
+			
+			appSettings.setS3SecretKey(newSettings.getS3SecretKey());
+			if (appSettings.getS3SecretKey().isEmpty()) {
+			   logger.warn("S3 recording is enabled but secret key is not given for app:{} ", getScope().getName());
+			}
+			
+			appSettings.setS3BucketName(newSettings.getS3BucketName());
+			if (appSettings.getS3BucketName().isEmpty()) {
+		   	   logger.info("S3 recording is enabled but bucket name is not given for app:{} ", getScope().getName());
+			}
+			
+			appSettings.setS3RegionName(newSettings.getS3RegionName());
+			if (appSettings.getS3RegionName().isEmpty()) {
+		  	  logger.info("S3 recording is enabled but region name is not given for app:{} ", getScope().getName());
+			}
+			
+			if (!"".equals(newSettings.getS3Endpoint())) {
```

Comment:
Checking content this way is better. It does not throw NullPointerException in any case. 
Another way is checking isEmpty method. 

It was `!newSettings.getS3Endpoint().equals("")`